Objects/bytesobject.c

```diff
@@ -1184,25 +1184,28 @@ PyObject *PyBytes_DecodeEscape(const char *s,
         unsigned char c = *first_invalid_escape;
         if ('4' <= c && c <= '7') {
             if (PyErr_WarnFormat(PyExc_DeprecationWarning, 1,
-                                 "invalid octal escape sequence '\\%.3s'",
-                                 first_invalid_escape) < 0)
+                                 "'\\%.3s' is an invalid escape sequence. "
+                                 "Such sequences will not work in the future. "
+                                 "Did you mean \"\\\\%.3s\"? A raw string is also an option.",
+                                 first_invalid_escape, first_invalid_escape) < 0)
```

Comment:
IMO, octal escapes in bytes should be handled differently, since the nature of the problem (the value is too large for a byte) and current resolution (the value wraps around) is fundamentally different.

Reply:
Could you elaborate further?

Reply:
Simply that an input like `b'\400'` is interpreted like `b'\x00'` (i.e. modulo 256), not like `b'\\400'` (assuming it wasn't intended to use an escape).

That said, in a `bytes` object it doesn't really make sense for an input like that to be intentional, since values above 255 are simply not representable. So on reflection it probably doesn't matter.

Reply:
I couldn't even find a use case to trigger this but maybe I didn't try hard enough. Nor the other bytes one. As you said, if you do `\400` you still get a valid output.

If you do `b'\o'` you get the same warning you get with '\o' regardless of whether or not I change the message in `bytesobject.c`.

Reply:
I don't think that that the hint "Did you mean ...?" is relevant here, I suggest to remove it.